enterprise/coderd/jfrog.go

```diff
@@ -0,0 +1,124 @@
+package coderd
+
+import (
+	"database/sql"
+	"encoding/json"
+	"net/http"
+
+	"github.com/google/uuid"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbauthz"
+	"github.com/coder/coder/v2/coderd/httpapi"
+	"github.com/coder/coder/v2/codersdk"
+)
+
+// Post workspace agent results for a JFrog XRay scan.
+//
+// @Summary Post JFrog XRay scan by workspace agent ID.
+// @ID post-jfrog-xray-scan-by-workspace-agent-id
+// @Security CoderSessionToken
+// @Accept json
+// @Produce json
+// @Tags Enterprise
+// @Param request body codersdk.JFrogXrayScan true "Post JFrog XRay scan request"
+// @Success 200 {object} codersdk.Response
+// @Router /exp/jfrog/xray-scan [post]
+func (api *API) postJFrogXrayScan(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	var req codersdk.JFrogXrayScan
+	if !httpapi.Read(ctx, rw, r, &req) {
+		return
+	}
+
+	payload, err := json.Marshal(req)
+	if err != nil {
+		httpapi.InternalServerError(rw, err)
+		return
+	}
```

Comment:
This payload includes the `workspaceID` and `agentID` which are columns in the table. We should omit those if we are going to write JSON to prevent having 2 sources for the same information.

If this struct is mostly static, why not just make all these columns?

Reply:
Mainly to not write database migrations. And honestly I just wanted to reuse the payload for fetch/post but I can make it more conventional.